src/encoding.ts

```diff
@@ -576,9 +582,9 @@ export function fieldDefs<F extends Field>(encoding: EncodingWithFacet<F>): Fiel
   return arr;
 }
 
-export function forEach<U extends Partial<Record<Channel, any>>>(
+export function forEach<U extends Record<any, any>>(
```

Comment:
Why go from the more strict type of `Partial<Record<Channel, any>>>` to the less strict `Record<any, any>`?

Reply:
I don't know exactly why but we actually get better type inference where we use `forEach`.